### DISCUSSION TITLE
Managing iOS build numbers

### DISCUSSION THREAD

**Discussion by ir-fuel** [fastlane/fastlane#17028] [Q&A]
Just wondering what the best way of going about this would be, considering the fact that due to 2FA restrictions it's impossible to get the build number from Appstore Connect.

I did this:
to create a new build, merge the code from `develop` in `staging`. In CircleCI I have a job that only runs on staging that executes my fastlane `betaRelease` lane.

That all works fine but the main issue is the build number. I wanted to simply get the latest build number from Appstore Connect, increase it and use it, but because of 2FA issues (check Github issues for that) this is not a stable option. 

So next I tried: increase the build number in the project file, build, push the increased build number to github. This however also has issues as:
- I have to commit to the same branch I am building from (which I don't want)
- Once I commit to that branch (staging), CircleCI is triggered again, and it starts building a new version of the app since `staging` has changed, and we effectively get into an infinite loop.

Is there another way to manage the build number somehow? I don't care that it's not in the project file. As long as it's always incremented while building it's fine.

**Comment by dekablade01** [comment]
The issues you mentioned above is not incorrect mechanism. 

You can commit to the `release/*` branch but the commit message must include [skip ci] for example 
REF: https://circleci.com/docs/2.0/skip-build/

```
[skip ci] set build build number to 8770
```


The new pushed commit won't be executed by the CircleCI. 

![Screen Shot 2020-08-11 at 11 37 38](https://user-images.githubusercontent.com/6483170/89857745-11184700-dbc7-11ea-9a44-e5a14fa10887.png)


-----------

You can ignore committing the build_number as well. But when you merge it back to `develop` or `master`. The build number in the tag, the build number in the binary and the build number in the project file won't be in sync.

**Reply by ir-fuel** [reply]
Thanks!
I just have one more question. Is there a way to merge that build number back to my `develop` branch somehow, from within fastlane? Otherwise the version number will not increment for the following builds and it will just do +1 on the one coming from development I think?

**Reply by dekablade01** [reply]
You have to do manually. 

You have to make a REST API Request to your Git provider to create a pull request. 

Right now, fastlane actions supports only github api. If you use Gitlab etc, you have to build it by your own. 

https://docs.fastlane.tools/actions/create_pull_request/

For Bitbucket, I created `create_bitbucket_pull_request` as a fastlane plugin. You can install it using fastlane plugin.

https://docs.fastlane.tools/plugins/available-plugins/

**Reply by ir-fuel** [reply]
I have a Github account. I just didn't see the pull request in the list of git actions.

**Reply by dekablade01** [reply]
I meant you have to use fastlane script to create a pull request using fastlane actions. Fastlane doesn't create it automatically for you. It depends on how you design the git-flow.

please check the fastlane document I left.

**Comment by joshdholtz** [comment]
Just wanted to put this here but...  2FA _shouldn't_ be a problem in the near-ish future. All of the APIs that use `Spaceship::ConnectAPI` and all of the tooling changes I'm working on will soon be able to use the JWT auth token which won't run into the 2FA issues. 

I should maybe get some information out on how to initially start testing / using that 🤔
